scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,188 @@
+require("scripts/globals/settings")
+require("scripts/globals/survival_guide_map")
+
+dsp = dsp or {}
+dsp.survivalGuide = dsp.survivalGuide or {}
+
+-- Determines if the survival guide teleport cost is like if you had a Rhapsody in White key item. Does not affect UI! (Default: 0)
+local SURVIVAL_GUIDE_TELEPORT_COST_GIL = 1000
+local SURVIVAL_GUIDE_TELEPORT_COST_TABS = 50
+
+-- This is used for the NationTeleport save/get
+local travelType = dsp.teleport.type.SURVIVAL
+local cutsceneID = 8500
+dsp.survivalGuide.expansions = 3 + (4 * ENABLE_COP) + (8 * ENABLE_TOAU) + (16 * ENABLE_WOTG) + (2048 * ENABLE_SOA)
+
+local optionMap =
+{
+    TELEPORT = 1,
+    UNKNOWN = 2,
+    SET_MENU_LAYOUT = 3,
+    ADD_FAVORITE = 4,
+    REMOVE_FAVORITE = 5,
+    REPLACE_FAVORITE = 6,
+    TELEPORT_MENU = 7
+}
+
+-------------------------------------------------
+-- local functions
+-------------------------------------------------
+
+local function checkForRegisteredSurvivalGuide(player, guide)
+    local group = guide.group
+    local hasRegisteredGuide = player:hasTeleport(travelType, guide.groupIndex - 1, guide.group - 1)
+
+    if not hasRegisteredGuide then
+        player:messageSpecial(zones[guide.zoneId].text.COMMON_SENSE_SURVIVAL)
+        player:addTeleport(travelType, guide.groupIndex - 1, guide.group - 1)
+
+        return false
+    end
+
+    return true
+end
+
+local function teleportMenuUpdate(player, option)
+    local choice = bit.band(option, 0xFF)
+
+    if choice >= optionMap.SET_MENU_LAYOUT and choice <= optionMap.TELEPORT_MENU then
+        local favorites = player:getTeleportMenu(travelType)
+
+        local index = bit.rshift(option, 16)
```

Comment:
```
ibm2431 4 days ago Contributor
Is this equivalent to bit.rshift(option, 16)?

Brian-Wuest 3 days ago Author
Fixed

Ivaar 11 hours ago Contributor
It's not the same, it was correct before.

ibm2431 6 hours ago Contributor
Was the idea to give the option padding bits on the right?

Ivaar 4 hours ago Contributor
Lua bit ops return signed 32 bit values. The lshift truncates the upper byte
so the next byte gets extracted by the shift right.

e.g.

bit.lshift(0x01020304, 8)
0x02030400
bit.rshift(0x02030400, 24)
0x02
i.e.
bit.rshift(bit.lshift(option, 8), 24)
is the same as
bit.rshift(bit.band(option, 0xFF0000), 16)
```

COpy and pasted form comment thread of the prior commit, as github isn't letting me reply to it.

I think 
`bit.rshift(bit.band(option, 0xFF0000), 16)`
is clearer

Reply:
Once we're collectively sure that bitshift is correct and that this has been tested in game by somebody, its merge ready I think.

Reply:
I verified that the change worked properly before committing.

I trusted what was suggested to me initially but I verified first 😸 .

Reply:
yeah, sadly I'll need someone to test after all the iterations you put in.